pkg/api/api.go

```diff
@@ -80,6 +88,35 @@ func (s *server) getOrCreateTag(tagUid string) (*tags.Tag, bool, error) {
 	return t, false, err
 }
 
+var errInvalidChunkAddress = errors.New("invalid chunk address")
+var errNoResolver = errors.New("no resolver connected")
+
+func (s *server) resolveNameOrAddress(str string) (swarm.Address, error) {
+	log := s.Logger
+
+	// Try and parse the name as a bzz address.
+	addr, err := swarm.ParseHexAddress(str)
+	if err == nil {
+		log.Tracef("name resolve: valid bzz address %q", str)
+		return addr, nil
+	}
+
+	// If no resolver is not available, return an error.
+	if s.Resolver == nil {
+		return swarm.ZeroAddress, errNoResolver
+	}
+
+	// Try and resolve the name using the provided resolver.
+	log.Debugf("name resolve: attempting to resolve %s to bzz address", str)
+	addr, err = s.Resolver.Resolve(str)
+	if err == nil && !addr.IsZero() {
+		log.Tracef("name resolve: resolved name %s to %s", str, addr)
+		return addr, nil
+	}
+
+	return swarm.ZeroAddress, fmt.Errorf("%w: %v", errInvalidChunkAddress, err)
```

Comment:
i don't understand this annotation, and i'm not sure why you are using one error to annotate another. also the assumption that the chunk address is invalid seems somewhat naive. `Resolve` cannot return other errors?